File: array_pop/array_pop.java
import java.util.Arrays;

public class array_pop {
    // ATOMIC_FUNCTION:  array_pop(arr)
    // TAGS: array
    // LANG:  java
    // FUNCTION: array_pop(arr)
    public Object[] array_pop(Object[] arr){
        int n=arr.length;
        Object[] result= new Object[n-1];
        System.arraycopy(arr, 0, result, 0, n-1);
        return result;


    }



    public static void main(String args[]) {
        String[] start_val = {"orange", "banana", "apple", "raspberry"};
        String[] expected_end_val={"orange", "banana", "apple"};
        array_pop a= new array_pop();
        Object[] end_val = a.array_pop(start_val);;


        // TEST
        if(Arrays.equals(expected_end_val,end_val)) {
            System.out.println("PASS");
        } else {
            System.out.println("FAIL");
        }
    }

}

File: array_pop/array_pop.js
// ATOMIC_FUNCTION: array.pop()
// TAGS: array
// LANG:  node.js
// FUNCTION: array.pop()

var start_val = ["orange", "banana", "apple", "raspberry"];
var expected_end_val = ["orange", "banana", "apple"];
start_val.pop();
var end_val = start_val;

// TEST
if(JSON.stringify(expected_end_val )==JSON.stringify(end_val)){
    console.log('PASS');
}else{
	console.log(end_val);
	
    console.log('FAIL');
}
